backend/src/main/java/org/eclipse/tractusx/puris/backend/masterdata/logic/service/MaterialServiceImpl.java

```diff
@@ -42,8 +44,30 @@ public class MaterialServiceImpl implements MaterialService {
     @Autowired
     private MaterialPartnerRelationService mprService;
 
+    @Autowired
+    private VariablesService variablesService;
+
     @Override
     public Material create(Material material) {
+        if(material.getOwnMaterialNumber() == null) {
+            log.error("Could not create material without ownMaterialNumber");
+        }
+        if(material.getMaterialNumberCx() == null) {
```

Comment:
To verify our common understanding: The materialNumberCx on material level always represents the catena-x id, that I would use to create the twin as a supplier while the materialPartnerRelationship catena-x id is the catena-x id I take over as a customer from the supplier's partTypeInformation aspect, right?

Neverthelesse we should also document that design decision in javadoc and in arc42 chapter 8 :)

Reply:
Yes, exactly